### PR TITLE
Made sure the editor can read the engine profiler content from a packaged version

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: editor/src/clj/editor/engine_profiler.clj [hunk 1]
```diff
@@ -51,14 +51,22 @@
...
+(defn- try-local-path [req]
   (let [path (str "." (:url req)) ; Make it a relative path as it starts with a "/"
-        mime-type (get-mime-type path)
-        resource (io/resource path)
-        data (if (= resource nil)
-               nil
-               (slurp resource))
-        code (if (= data nil) 404 200)]
+        resource (io/resource path)]
+    resource))
```
Comment (JCash) [inline editor/src/clj/editor/engine_profiler.clj:57]: If you test locally, the files should be local: "./path/to/file"

File: editor/src/clj/editor/engine_profiler.clj [hunk 2]
```diff
@@ -51,14 +51,22 @@
...
+
+(defn- try-package-path [req]
+  (let [path (subs (:url req) 1) ; remove the initial "/"
```
Comment (JCash) [inline editor/src/clj/editor/engine_profiler.clj:60]: If you're loading directly from the package, the paths should be like: "path/to/file"

**Review by AGulev — DISMISSED** [review]
LGTM, but it would be better if Vlad takes a look.

File: editor/src/clj/editor/engine_profiler.clj [hunk 3]
```diff
@@ -51,14 +51,22 @@
...
+        resource (io/resource path)]
+    resource))
+
+(defn- read-resource [req] 
+  (let [mime-type (get-mime-type (:url req))
+        resource (try-package-path req)
+        resource (if (nil? resource) (try-local-path req) resource)
```
Comment (vlaaad) [inline editor/src/clj/editor/engine_profiler.clj:67]: Can be simplified, IMO this is easier to understand:
```suggestion
        resource (or (try-package-path req) (try-local-path req))
```

File: editor/src/clj/editor/engine_profiler.clj [hunk 4]
```diff
@@ -51,14 +51,22 @@
...
+        data (if (nil? resource) nil (slurp resource))
```
Comment (vlaaad) [inline editor/src/clj/editor/engine_profiler.clj:68]: Minor: `some->` is a common idiom for transforming data with a null check.
```suggestion
        data (some-> resource slurp)
```

File: editor/src/clj/editor/engine_profiler.clj [hunk 5]
```diff
@@ -51,14 +51,22 @@
...
+        code (if (nil? data) 404 200)]
```
Comment (vlaaad) [inline editor/src/clj/editor/engine_profiler.clj:69]: Minor: IMO easier to understand. 
```suggestion
        code (if data 200 404)]
```

**Review by vlaaad — DISMISSED** [review]
LGTM! I added some suggestions to improve readability.

File: editor/src/clj/editor/engine_profiler.clj (hunk 5 above)
Comment (JCash) [reply editor/src/clj/editor/engine_profiler.clj:69]: Yeah, not sure why I didn't get it to work the first time. It works now 🤷

**Pull request merged**